rfc/rfc-79/rfc-79-2.md

```diff
@@ -0,0 +1,99 @@
+* Have the ingestion writer write a "hint" of possible partitions it might affect in the .requested file, and the cancellable table service writer can check that before commiting the table service plan
+* If the cancellable table service writer sees that there is a .requested file for an ingestion action, it can try to wait some time for the .inflight to appear before performing write reconcilation checks
+
+### Handling cancellation of plans
+An additional config "cancellation-policy" can be added to the table service plan to indicate when it is ellgible to be permenatnly rolled back by writers other than the one responsbible for executing the table service. This policy can be a threshold of hours or instants on timeline, where if that # of hours/instants have elapsed since the plan was scheduled, any writer/operation can target it for rollback via clean. This policy should be configured by the writer scheduling a cacnellable table service, based on the amount of time they expect the plan to remain on the timeline before being picked up for execution. For example, if a table service writer is expected to immeditately start executing the plan after scheduling it, the the cancellation-policy can just be a few minutes. On the other hand, if the plan is expected to have its execution deferred to a few hours later, then the cancellation-policy should be more lenient. Note that this cancellation policy is not a repalacement for determining wether a table service plan is currently being executed - as wtih ingestion writes, cleanup of a cancellable table service plan should only start once it is confirmed that a ongoing writer is no longer progressing it. 
+
+In order to ensure that other writers can indeed permenantely cancel a cancellable table service plan (such that it can no longer be executed), additional changes to clean and table service flow will be need to be added as well. Two proposals are detailed below. Also, note that the cancellation-policy is only required to be honored by clean: a user can choose setup an application to aggresively clean up a failed cancellable table service plan even if it has not meet the critera for its cancellation-policy yet. This can be useful if a user wants a utility to manually ensure that clean/archival for a dataset progresses immdeitately or knows that a cancellable table service plan will not be attempted again or cleaned up by another writer. Each proposal provides an example on how to achieve this.
+
+#### (A) Making cancellable plans "mutable" 
```

Comment:
I guess we are not calling out another drawback of this approach. 
a concurrent table service writer, could encounter FileNotFound issue while accessing the timeline if another writer rollsback the table service of interest and by removing all meta files from the timeline. 

Don't we need to handle this ?

Reply:
Yes mutable plan approach has this issue. But this is anyway an existing issue https://issues.apache.org/jira/browse/HUDI-8138 that can happen right now (with insert_overwrite replacecommits) so I wasn't sure wether to explicitly add this.